source/agora/node/Registry.d

```diff
@@ -475,8 +496,8 @@ public class NameRegistry: NameRegistryAPI
 
     ***************************************************************************/
 
-private PublicKey parsePublicKeyFromDomain (in char[] domain,
-    ZoneData zone) @safe
+private PublicKey parsePublicKeyFromDomain (string TemplateRegistry = "") (in char[] domain,
+    ZoneData!(TemplateRegistry) zone) @safe
```

Comment:
Personally I am not too keen on templating a `struct` based on some internal variable. It always comes back to bite you, exactly like it does here. Since D has a nominal type system, you loose substitution for what amounts to the value of a variable.

But if you really want to go with this approach, a better way would be to template the whole `zone` type, e.g.
```D
private PublicKey parsePublicKeyFromDomain (ZoneDataT) (in char[] domain,
    ZoneDataT zone) @safe
```

And let the compiler infer it. You *can* get fancy and put template constraints, either by doing deconstruction:
```D
private PublicKey parsePublicKeyFromDomain (string TableName, ZoneDataT : ZoneData!TableName ) (in char[] domain,
    ZoneDataT zone) @safe
```

(Something like this... Not tested, but it definitely can be expressed) or with constraints:
```D
private PublicKey parsePublicKeyFromDomain (ZoneDataT) (in char[] domain,
    ZoneDataT zone) @safe
    if (is(ZoneDataT : ZoneData!TableName, TableName)
```

But if you're going that far, might as well pass `TableName` as a runtime value to the constructor, OR hide the problem by making the method a member of `ZoneData`.